ipa-core/src/ff/boolean_array.rs

```diff
@@ -497,6 +497,28 @@ macro_rules! boolean_array_impl {
                 }
             }
 
+            /// This function converts an `BA` type into a vector of `32 bit` Galois field elements.
+            ///
+            /// ##Errors
+            /// Outputs an error when conversion from raw slice to Galois field element fails.
+            impl TryFrom<$name> for Vec<Gf32Bit> {
+                type Error = crate::error::Error;
+
+                fn try_from(value: $name) -> Result<Self, Self::Error> {
+                    // len() returns bits, so divide by 8
+                    // further divide by 4 since Gf32Bit has 4 byte
+                    // add 31 to round up
+                    let length = (value.0.len() + 31) / 32;
+                    let mut chunks = Vec::<Gf32Bit>::with_capacity(length);
+                    let last_chunk_start = length-1;
```

Comment:
the formatting here is messed up quite a bit, can you take that function out and run `cargo fmt` before taking it back inside the macro?